### PR TITLE
Enhance the Build Environment chapter of the user manual

### PR DESCRIPTION
**Pull request by eskatos**
This PR enhances the Build Environment chapter of the user manual with more detailled snippets demonstrating current behaviors and DSLs for using Gradle properties, System properties, environment variables and project properties.

The documentation now exhibits the similarities and differences in the Groovy and Kotlin DSLs.

Preview is available at https://builds.gradle.org/repository/download/Gradle_Master_Check_BuildDistributions/52992331:id/distributions/gradle-7.6-docs.zip!/gradle-7.6-20220616074312%2B0000/docs/userguide/build_environment.html

Along the way, the samples testing infrastructure has been enhanced to allow sample tests to define environment variables as this was required in order to test the added snippets.

### CODE REVIEW

**Comment by eskatos** [comment]
@bot-gradle test BD

File: subprojects/docs/src/docs/userguide/running-builds/build_environment.adoc [hunk 1]
```diff
@@ -24,10 +24,10 @@ When configuring Gradle behavior you can use these methods, listed in order of h
 
 * <<command_line_interface.adoc#command_line_interface, Command-line flags>> such as `--build-cache`. These have precedence over properties and environment variables.
 * <<#sec:gradle_system_properties, System properties>> such as `systemProp.http.proxyHost=somehost.org` stored in a `gradle.properties` file.
```
Comment (abstratt) [inline subprojects/docs/src/docs/userguide/running-builds/build_environment.adoc:26]: Not from your changes, but the first two times we mention this convention, we omit the fact that it only works for the root project

Comment (abstratt) [reply subprojects/docs/src/docs/userguide/running-builds/build_environment.adoc:26]: https://github.com/gradle/gradle/pull/21006/files#diff-bf47e49178bd8fc0fbf491777da11ff52fbb8b8120da0368064b71fad6cc371cR194 may be the best place to male that clear.

**Comment by abstratt** [comment]
Looks good to me, but you may want another opinion...

File: subprojects/docs/src/docs/userguide/running-builds/build_environment.adoc (hunk 1 above)
Comment (eskatos) [reply subprojects/docs/src/docs/userguide/running-builds/build_environment.adoc]: Good point, thanks! And fixed

**Comment by eskatos** [comment]
@bot-gradle test BD

**Comment by eskatos** [comment]
@bot-gradle test and merge

**Pull request merged**